Comprehension/use spelling rule feedback

anathomical opened the pull request:
## WHAT
Pull Spelling feedback from the universal "Spelling" rule in the Comprehension database rather than hard-coding a constant bit of feedback.
## WHY
So that the Curriculum team has more control over what students see in order to maximize the value of the Feedback student get.
## HOW
Leave the constant feedback string as a fallback, but update the code to pull feedback text from the Feedback records associated with the universal Spelling Rule when it is available.

### Notion Card Links
https://www.notion.so/quill/Changing-spelling-feedback-in-the-Universal-Rules-page-doesn-t-change-the-feedback-in-the-student-ap-e104a79991784650962840d44a71484d

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  Yes
Have you deployed to Staging? | NO - tiny change
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Design Review: If applicable, have you compared the coded design to the mockups? | N/A

anathomical committed:
Update schema file to capture minor migration

anathomical committed:
Update Comprehension Spelling API to pull Feedback from universal rule

brendanshean commented on services/QuillLMS/engines/comprehension/lib/comprehension/comprehension/spelling_check.rb:
```diff
@@ -27,13 +27,17 @@ def feedback_object
       bing_response['error'] ? bing_response['error']['message'] : nil
     end
 
     private def spelling_rule
```
Would memoization apply here?
`@spelling_rule ||= Rule.where(rule_type: FEEDBACK_TYPE).first`

anathomical replied:
Sure, that seems reasonable.

brendanshean reviewed (approved):
One minor comment but looks good.

anathomical committed:
Use memoization based on PR feedback

anathomical committed:
Tweak test to make it deterministic

There was an edge case where a factory sequence (from "Name 9" to "Name 10" or similar) could result in alphabetical ordering unexpectedly putting the later-created item ahead of the earlier-created item.
